src/vis-preview.py

```diff
@@ -42,39 +43,84 @@ def main():  # pragma: no cover
     parser.add_argument(
         '--to_json', action='store_true',
         help='Output viewconf, rather than open in browser.')
+    # parser.add_argument(
+    #     '--parent_uuid', action='store_true',
+    #     help='Parent uuid for the dataset',
+    #     default=None)
+    parser.add_argument(
+        '--epic_uuid', metavar='UUID',
+        help='uuid of the EPIC dataset',
+        default=None)
+
+    #
+    # parser.add_argument(
+    #     '--epic_builder',  action='store_true',
+    #     help='Whether to use the epic_builder or not',
+    #     default=None)
 
     args = parser.parse_args()
     marker = args.marker
+    # epic_builder = args.epic_builder
+    epic_uuid = args.epic_uuid
+    # parent_uuid = args.parent_uuid # this may not be needed, as the --url provides the parent dataset json?
 
     if args.url:
         response = requests.get(args.url)
         if response.status_code == 403:
-            # Even if the user has provided a globus token,
-            # that isn't useful when making requests to our portal.
             raise Exception('Protected data: Download JSON via browser; Redo with --json')
```

Comment:
Maybe we could re-try the request with the token included, if one has been provided?